components/validator/guardian.go

```diff
@@ -0,0 +1,159 @@
+package validator
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	_ "net/http/pprof"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/kroma-network/kroma/bindings/bindings"
+	"github.com/kroma-network/kroma/components/node/eth"
+	"github.com/kroma-network/kroma/utils"
+)
+
+// Guardian is responsible for validating outputs
+type Guardian struct {
+	log      log.Logger
+	cfg      Config
+	ctx      context.Context
+	cancel   context.CancelFunc
+	callOpts *bind.CallOpts
+	txOpts   *bind.TransactOpts
+	wg       sync.WaitGroup
+
+	securityCouncilContract *bindings.SecurityCouncil
+	securityCouncilSub      ethereum.Subscription
+}
+
+// NewGuardian creates a new Guardian
+func NewGuardian(parentCtx context.Context, cfg Config, l log.Logger) (*Guardian, error) {
+	securityCouncilContract, err := bindings.NewSecurityCouncil(cfg.SecurityCouncilAddr, cfg.L1Client)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx, cancel := context.WithCancel(parentCtx)
+
+	return &Guardian{
+		log:                     l,
+		cfg:                     cfg,
+		ctx:                     ctx,
+		cancel:                  cancel,
+		callOpts:                utils.NewCallOptsWithSender(ctx, cfg.TxManager.From()),
+		txOpts:                  utils.NewSimpleTxOpts(ctx, cfg.TxManager.From(), cfg.TxManager.Signer),
+		securityCouncilContract: securityCouncilContract,
+	}, nil
+}
+
+func (g *Guardian) Stop() {
+	g.log.Info("stop Guardian")
+
+	if g.securityCouncilSub != nil {
+		g.securityCouncilSub.Unsubscribe()
+	}
+}
+
+func (g *Guardian) handleValidationRequested(logs chan *bindings.SecurityCouncilValidationRequested) {
+	defer func() {
+		g.log.Info("exit handleRequestValidated goroutine")
+		close(logs)
```

Comment:
Below link says it's good to **CLOSE CHANNEL ON THE SENDER SIDE**. Could we make a rule for this? cc @kroma-network/l2-protocol 

Please see https://go101.org/article/channel-closing.html.

>One general principle of using Go channels is don't close a channel from the receiver side and don't close a channel if the channel has multiple concurrent senders. In other words, we should only close a channel in a sender goroutine if the sender is the only sender of the channel.

Reply:
FYI, @0xHansLee `l2OutputSubmittedEventChan` and `challengeCreatedEventChan` in `challenger.go` seems not closed.